File: src/Moq/Moq.DynamicProxy/Mock.DynamicFactory.cs
namespace Moq
{
    partial class Mock
    {
        static Mock()
        {
            Sdk.MockFactory.Default = new Sdk.DynamicMockFactory();
            OnInitialized();
        }

        /// <summary>
        /// Invoked after the default <see cref="MockFactory.Default"/> 
        /// is initialized.
        /// </summary>
        static partial void OnInitialized();
    }
}

File: src/Moq.DynamicProxy/Mock.DynamicFactory.vb
Namespace Global.Moq
    Partial Friend Class Mock
        Shared Sub New()
            Sdk.MockFactory.[Default] = New Sdk.DynamicMockFactory
            OnInitialized()
        End Sub

        ''' <summary>
        ''' Invoked after the default <see cref="MockFactory.Default"/> 
        ''' is initialized.
        ''' </summary>
        Partial Private Shared Sub OnInitialized()

        End Sub
    End Class
End Namespace